cpp/src/fil/infer.cu

```diff
@@ -282,43 +274,34 @@ struct tree_aggregator_t<NITEMS, TREE_PER_CLASS_MANY_CLASSES> {
         (vec<NITEMS, float>*)((char*)shared_workspace + data_row_size)),
       num_classes(num_classes_) {
     for (int c = threadIdx.x; c < num_classes; c += blockDim.x)
-      per_class_margin[c].fill(0.0f);
-    //__syncthreads(); // done in the main loop
+      per_class_margin[c] = vec<NITEMS, float>();
```

Comment:
Does this initialize the per-class margins with 0?

Reply:
yes. See https://en.cppreference.com/w/cpp/language/value_initialization "4) otherwise, the object is zero-initialized."